clientconn_state_transition_test.go

```diff
@@ -396,7 +400,9 @@ func (s) TestStateTransitions_MultipleAddrsEntersReady(t *testing.T) {
 			return
 		}
 
-		<-sawReady
+		if _, closed := <-sawReady; closed {
```

Comment:
Hi, I am not sure if my understanding is correct. Do you mean that we can leave this line as `<-sawReady`, and when the parent goroutine exits by `t.Fatal()`, we still let the child goroutine executes `<-sawReady` and then `conn.Close()`? 
I think this is fine. I can change this line back to `<-sawReady` if you agree

Reply:
I don't see why we are calling `Accept()` here? (after reading from `sawReady` and calling `conn.Close()`)

I think it should be safe just to do the following here:
```golang
  <-sawReady
  conn.Close()
  close(server1Done)
```
Please let me know if you think otherwise.